src/context/json.ts

```diff
@@ -1,11 +1,17 @@
 import { ResponseTransformer } from '../response'
 
+type JsonArray = (JsonValue | JsonRecord)[]
+type JsonRecord = Record<string, unknown>
+type JsonValue = string | boolean | number | JsonRecord | JsonArray | null
```

Comment:
This looks pretty good! I like the idea of being this strict, but it seems like we should consider matching `JSON.stringify`'s type of `any` here. 

The only gotcha I can think of here with these types as-is is that this would potentially force a TypeScript user to transform `Date`s before being passed in. I don't personally think that's really a problem but wanted to bring it up for @kettanaito's review.

Reply:
Oops! Definitely should support Date since the method converts it to a string.

Reply:
I agree with this because is what `JSON.stringify` does 

```js
/**
* Converts a JavaScript value to a JavaScript Object Notation (JSON) string.
* @param value A JavaScript value, usually an object or array, to be converted.
* @param replacer An array of strings and numbers that acts as a approved list for selecting the object properties that will be stringified.
* @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
*/
stringify(value: any, replacer?: (number | string)[] | null, space?: string | number): string;
```

Reply:
I agree with @msutkowski on typing the value as `any`, just as `JSON.stringify` has. That leaves the type annotation aligned with the underlying stringification, and reduces the cost of maintaining it in the future. If an invalid value is passed, an error will be raised when stringifying it. It's the end developer's responsibility to pass a valid JSON to `ctx.json()`.